commcare_connect/reports/views.py

```diff
@@ -106,14 +116,92 @@ def payment_strings(payment_data):
     }
 
 
-@login_required
-@user_passes_test(lambda user: user.is_superuser)
-@require_GET
-def delivery_stats_report(request):
-    table_data = []
-    quarters = _get_quarters_since_start()
-    for q in quarters:
-        data = _get_table_data_for_quarter(q)
-        table_data.append(data)
-    table = AdminReportTable(table_data)
-    return render(request, "reports/admin.html", context={"table": table})
+class SuperUserRequiredMixin(LoginRequiredMixin, UserPassesTestMixin):
+    def test_func(self):
+        return self.request.user.is_superuser
+
+
+class DeliveryReportFilters(django_filters.FilterSet):
```

Comment:
It seems like we are using `django-filters` in a pretty nontraditional way, based on all the overrides (`filter_by_none`, `NonModelFilterView`, etc), what functionality are we getting from it?

Reply:
This was a bit of a non-traditional filter (in that it's not based on model fields), you can how useful it is in the Events PR https://github.com/dimagi/commcare-connect/pull/349